---
title: "05_analysis_1"
author: "Group_23"
format:
  html:
    embed-resources: true
editor: visual
---

# 07. Differential expression analysis

```{r}
#| eval: true 
#| message: false
#| warning: false
library("tidyverse")
library("broom")
library("ggrepel")
library("rentrez")
```

## 07.1. Introduction

In the present quarto document a Differential Expression Analysis of the available proteomics data is conducted. This kind of analysis allows to identify changes in the relative abundance of proteins between patients that present different conditions. In the present notebook two different conditions which were considered of interest will be studied:

-   Triple-Negative Breast Cancer (TNBC) vs. Non-TNBC

-   Cancer in nearby lymph nodes vs. No-Cancer in nearby lymph nodes

This analysis is of special importance for understanding the underlying mechanisms of the disease, and it is also useful to identify proteins which can be used as bio-markers for disease diagnostics or prognosis, as the expression fo certain proteins may be associated with a special variant of the disease or a better or worse prognosis.

As it was decided to test two conditions, and the code required for this analysis is fairly similar just taking into consideration the condition, it was decided to create several functions which can be accessed in the R script `99_proj_func.R`. More specifically, the functions are:

-   `DEA_proteins`: takes as input the tibble with the data, and the column which is the condition to be tested (has to be called inside a `dplyr` pipeline). In this function the relevant columns for the analysis are selected and a nested data structure is created to fit a linear model for each of the proteins. The models are tidied afterwards, and the p-values are adjusted. This function calls the function `volcano_plot`, which creates a volcano plot with the results from the analysis. The output of the function is a list that includes a tibble with the results of the analysis and a volcano plot. Further description of the different steps can be found in the comments of the function.

-   `volcano_plot:` creates a volcano plot with the data from the previous function. The Non Significant proteins are plotted in grey, the over-expressed in red and the under-expressed in blue.

```{r}
# Getting the functions from the code
source("99_proj_func.R")
```

## 07.2. Load data from .tsv files

```{r}
data <- read.delim("../data/03_dat_aug.tsv") |>
  as_tibble()
```

## 07.3. Triple-Negative Breast Cancer (TNBC) vs. Non-TNBC

As it was already explained during the description of the data, in this data set the relative abundance of proteins in the samples is represented by means of Log2 iTRAQ ratios were used to represent the relative abundance of proteins in the samples.

In the context of Breast Cancer studies it is of special interest the study of Triple-Negative Breast Cancer (TNBC), which are characterized for lacking the expression of the estrogen receptor (ER), progesterone receptor (PR), and human epidermal growth factor receptor 2 (HER2). This type of cancer is usually more aggressive and additionally, there are no approved targeted treatments available, and even if they can be treated with chemotherapy, patients that suffer from this cancer are more prone to early relapse and metastasis. Therefore further research on the differences between these type of cancer and others is of special interest.

To proceed with the analysis the function DEA_proteins will be used, in this case using the variable TNBC as the condition to be tested.

```{r}
DEA_TNBC <- data |>
  DEA_proteins(TNBC)

```

From the element estimates of the list, a tibble is obtained. It is possible to check how many proteins are differentially expressed (significant after multiple test correlation):

```{r}
DEA_TNBC |>
  pluck("estimates") |>
  count(dif_exp)
```

There are 199 proteins differentially expressed when testing TNBC vs. Non-TNBC. 115 proteins with a higher expression in patients that are TNBC than in patients that are not TNBC, and 84 with a lower expression in patients that are TNBC than in patients that are not TNBC.

We can visualize the volcano plot corresponding to this test:

```{r}
DEA_TNBC |>
  pluck("estimates")
```

```{r}
DEA_TNBC |>
  pluck("plt_volcano")
```

```{r}
#| eval: true
#| echo: false
ggsave("../results/TNBC_vs_Non_TNBC.png", plot = last_plot(), device = "png",bg="white")
```

Now, get a tibble with all the differentially expressed proteins (Up or Down) for further comparison.

```{r}
TNBC_dif_exp <- DEA_TNBC |>
  pluck("estimates") |>
  filter(dif_exp == "Up" | dif_exp == "Down") |>
  select(Protein,
         dif_exp)
```

## 07.4. Cancer in nearby nodes vs. Non-cancer in nearby nodes

The variable Node_Cancer refers to the presence of cancer in the nearby lymph nodes. It is 1 if there is cancer in the nearby nodes, this can be different number and location of lymph nodes with cancer, or it can take the value 0 if there is not. Differential expression analysis of proteins between cancer patients with cancer in the nearby lymph nodes, and those without, can provide insights into the underliyng molecular mechanisms which are involved in Breast Cancer development and the spread of it to other lymph nodes. The results could be used with regards to diagnosis, as for bio-markers identification, or event for treatment.

To proceed with the analysis the function DEA_proteins will be used, in this case using the variable Node_Cancer as the condition to be tested.

```{r}
DEA_nodes <- data |>
  DEA_proteins(Node_Cancer)
```

```{r}
DEA_nodes |>
  pluck("estimates") |>
  count(dif_exp)
```

We can see how the majority of proteins are not differentially expressed between patients with cancer in nearby lymph nodes and patients that do not present cancer in the nearby lymph nodes. There is only one gene which is significantlly under-expressed in patients that have cancer in the nearby lymph nodes, therefore it could be interesting to take that protein into account for further analysis.

We can visualize the volcano plot corresponding to this test:

```{r}
DEA_nodes |>
  pluck("plt_volcano") 
```

```{r}
#| eval: true
#| echo: false
ggsave("../results/NODE_vs_Non_NODE.png", plot = last_plot(), device = "png",bg="white")
```

Now, get a tibble with the name of the protein which is differentially expressed.

```{r}
NODE_dif_exp <- DEA_nodes |>
  pluck("estimates") |> 
  filter(dif_exp == "Up" | dif_exp == "Down") |>
  select(Protein,
         dif_exp)
```

## 07.5. Comparing differentially expressed proteins in both tests

Given that TNBC cancer are known for its aggressive nature and the fact that cancer in nearby nodes is a conditions which can be related to poor cancer outcomes, it can be interesting to check if the proteins which is under-expressed in patients with cancer in nearby nodes is also under-expressed in patients with TNBC.

Compare list of genes for the comparisons that have been made:

```{r}
NODE_dif_exp |>
  inner_join(TNBC_dif_exp, by = c("Protein","dif_exp")) |>
  count(dif_exp)
```

Therefore this proteins is not under-expressed in patients with TNBC. Could it be possible that it is over-expressed?

```{r}
NODE_dif_exp |>
  inner_join(TNBC_dif_exp, by = c("Protein")) 
```

Therefore there are not significant differences in the expression of this proteins between patients with TNBC and patients without TNBC.

The package `rentrez` allows to retrieve data from NCBI's databases. In this case, first the RefSeq IDs are retrieved, and the the function `entrez_fetch` is used to retrieve information corresponding to those IDs. This information retrieved is parsed to get only the information corresponding to the brief description of the protein, regex is used for this purpose. This allows to get an overview of the role of these proteins. In this case the proteins checked is the protein which is differentially expressed in the nearby cancer in nodes test, and the first 6 proteins of TNBC which are not in Node (this is to avid excessive run time, as there are 199 proteins, but this code could be reused for more meaningful comparisons).

```{r}
# For proteins in NODE that are not in TNBC
node_not_tnbc <- NODE_dif_exp |>
  anti_join(TNBC_dif_exp, by = c("Protein", "dif_exp")) |>
  select(Protein) |>
  mutate(Info = map(Protein, ~ sub("^>(.*?)\\[Homo.*", "\\1", entrez_fetch(db = "protein", id = .x, rettype = "fasta"))))


# For proteins in TNBC that are not in NODE
tnbc_not_basal <- TNBC_dif_exp |>
  anti_join(NODE_dif_exp, by = c("Protein","dif_exp")) |>
  head() |>
  select(Protein) |>
  mutate(Info = map(Protein, ~ sub("^>(.*?)\\[Homo.*", "\\1", entrez_fetch(db = "protein", id = .x, rettype = "fasta"))))
```

```{r}
node_not_tnbc |>
  select(Info) |>
  pull()
```

```{r}
tnbc_not_basal |>
  select(Info) |>
  pull()
```

## 07.6. Conclusion

The study of differential expression of proteins is crucial for several reasons, among them that it allows to identify which signature proteins are associated with each subtype of cancer, which could shed light into the molecular differences which lead cancer development and progressions. It could potentially be useful regarding the different therapies that can be developed for cancer treatment, as some of these proteins could potentially be therapeutic targets or new biomarkers.

In the present analysis it has been proven that there is a difference in the protein expression profile of patients with TNBC and patients that do not have TNBC. On the other hand, almost none difference in the protein expression profile have been found between patients that develop cancer in the nearby lymph node and those who do not. Only one protein is differently expressed between these two sets of patients. Furthermore, there are not significant differences in the expression of this proteins between patients with TNBC and patients without TNBC. This indicates that it does not seem to be any kind of relationship between these conditions which have been tested.

Further research on the function of these proteins could shed light in the biological and molecular mechanisms which lead cancer development in one fo the most aggressive cancers which is TNBC. Additionally, the protein which seems to be under-expressed in patients with no cancer in nearby nodes could be a potential biomarker or therapeutic target. With this regard, further research should focus on Gene Set Enrichment Analysis, which could allow to get a more comprehensive view of the classes of proteins over-represented in these sets of proteins.
